kyverno/kyverno
# fix: ur not cleaned up when deleting namespaced policy

Signed-off-by: Sandesh More <sandesh.more@infracloud.io>

## Explanation
Currently same handler function is used in both ```ClusterPolicy``` and ```Policy``` deletion, due to which a type error prevents clean up of UR while policy deletion.

## Related issue

closes: 5101

## Milestone of this PR


## What type of PR is this
bug

## Proposed Changes
Added seperate handler for Namespaced Policy deletion which takes care of UR clean up.

### Proof Manifests
1. Create the poltest Namespace.
2. Create this Policy.
```
apiVersion: kyverno.io/v2beta1
kind: Policy
metadata:
  name: pol-sync-data
  namespace: poltest
spec:
  rules:
  - name: gen-zk
    match:
      any:
      - resources:
          kinds:
          - Secret
    generate:
      synchronize: true
      apiVersion: v1
      kind: ConfigMap
      name: zk
      namespace: poltest
      data:
        kind: ConfigMap
        metadata:
          labels:
            somekey: somevalue
        data:
          ZK_ADDRESS: "192.168.10.10:2181,192.168.10.11:2181,192.168.10.12:2181"
          KAFKA_ADDRESS: "192.168.10.13:9092,192.168.10.14:9092,192.168.10.15:9092"
```
3. Create a Secret which triggers this rule.
 ```$ k -n poltest create secret generic noodles --from-literal=foo=bar```
4. Delete the Policy.
```$ k -n poltest delete pol pol-sync-data```
6. See the complete UR is removed: 
```$ k -n kyverno get ur ```

## Checklist



  
  
  
  

## Further Comments
